tools/purge_trigger_instances.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+A utility script that purges trigger instances older than certain
+timestamp.
+
+*** RISK RISK RISK. You will lose data. Run at your own risk. ***
+"""
+
+from datetime import datetime
+import pytz
+import sys
+
+from oslo_config import cfg
+
+from st2common import config
+from st2common import log as logging
+from st2common.script_setup import setup as common_setup
+from st2common.script_setup import teardown as common_teardown
+from st2common.persistence.trigger import TriggerInstance
+from st2common.util import isotime
+
+LOG = logging.getLogger(__name__)
+DELETED_COUNT = 0
+
+
+def _do_register_cli_opts(opts, ignore_errors=False):
+    for opt in opts:
+        try:
+            cfg.CONF.register_cli_opt(opt)
+        except:
+            if not ignore_errors:
+                raise
+
+
+def _register_cli_opts():
+    cli_opts = [
+        cfg.StrOpt('timestamp', default=None,
+                   help='Will delete trigger instances older than ' +
+                   'this UTC timestamp. ' +
+                   'Example value: 2015-03-13T19:01:27.255542Z')
+    ]
+    _do_register_cli_opts(cli_opts)
+
+
+def _purge_model(instance):
+    try:
+        TriggerInstance.delete(instance)
+    except:
+        LOG.exception('Exception deleting instance %s.', instance)
+    else:
+        global DELETED_COUNT
+        DELETED_COUNT += 1
+
+
+def purge_trigger_instances(timestamp=None):
+    if not timestamp:
+        LOG.error('Specify a valid timestamp to purge.')
+        return
+
+    LOG.info('Purging trigger instances older than timestamp: %s' %
+             timestamp.strftime('%Y-%m-%dT%H:%M:%S.%fZ'))
+
+    # XXX: Think about paginating this call.
+    filters = {'occurrence_time__lt': isotime.parse(timestamp)}
+    instances = TriggerInstance.query(**filters)
```

Comment:
Yeah, we had this discussion last time. Some profiling here is in order. I'll take this up in a subsequent PR.